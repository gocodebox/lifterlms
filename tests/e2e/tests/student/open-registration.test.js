/**
 * Test the Setup Wizard
 *
 * @since [version]
 */

const { visitAdminPage } = require( '@wordpress/e2e-test-utils' );

const {
	clickAndWait,
	fillField,
	logoutUser,
	visitPage,
} = require( 'llms-e2e-test-utils' );

let openRegStatus = null;

/**
 * Toggles the open registration setting on or off
 *
 * @since  [version]
 *
 * @param  {Boolean} status Whether to toggle on (`true`) or off (`false`).
 * @return {void}
 */
const toggleOpenReg = async function( status ) {

	if ( openRegStatus === status ) {
		return;
	}

	// Launch the Setup Wizard.
	await visitAdminPage( 'admin.php', 'page=llms-settings&tab=account' );

	const curr_status = await page.$eval( '#lifterlms_enable_myaccount_registration', el => el.checked );

	if ( status && ! curr_status || ! status && curr_status ) {
		await page.click( '#lifterlms_enable_myaccount_registration' );
		await clickAndWait( '.llms-save .llms-button-primary' );
		openRegStatus = status;
	}

}

describe( 'OpenRegistration', () => {

	afterEach( async () => {
		await logoutUser();
	} );

	it ( 'should not allow registration because user is already logged in.', async () => {

		await toggleOpenReg( true );
		await visitPage( 'dashboard' );
		await expect( await page.$( '.llms-new-person-form-wrapper > h4.llms-form-heading' ) ).toBeNull();

	} );

	it ( 'should allow registration.', async () => {

		await toggleOpenReg( true );
		await logoutUser();
		await visitPage( 'dashboard' );
		expect( await page.$eval( '.llms-new-person-form-wrapper > h4.llms-form-heading', el => el.textContent ) ).toBe( 'Register' );

	} );

	it ( 'should register a new user.', async () => {

		await toggleOpenReg( true );
		await logoutUser();

		await visitPage( 'dashboard' );

		const
			the_int = Math.floor( Math.random() * ( 99990 - 10000 + 1 ) ) + 10000,
			email   = 'help+e2e' + the_int + '@lifterlms.com',
			pass    = Math.random().toString( 36 ).slice( 2 ) + Math.random().toString( 36 ).slice( 2 );

		await fillField( '#email_address', email );
		await fillField( '#password', pass );
		await fillField( '#password_confirm', pass );
		await fillField( '#first_name', 'Jeffrey' );
		await fillField( '#last_name', 'Lebowski' );

		await clickAndWait( '#llms_register_person' );

		expect( await page.$eval( 'h2.llms-sd-title', el => el.textContent ) ).toBe( 'Dashboard' );

	} );

	it ( 'should not allow registration because open registration is disabled.', async () => {

		await toggleOpenReg( false );
		await logoutUser();
		await visitPage( 'dashboard' );
		await expect( await page.$( '.llms-new-person-form-wrapper > h4.llms-form-heading' ) ).toBeNull();

	} );

} );
