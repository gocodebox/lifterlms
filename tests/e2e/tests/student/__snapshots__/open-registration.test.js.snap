// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpenRegistration Localization should localize city, state, and postcode fields when changing the selected country 1`] = `"Province*"`;

exports[`OpenRegistration Localization should localize city, state, and postcode fields when changing the selected country 2`] = `
Object {
  "AH": "Anhui",
  "BJ": "Beijing",
  "CQ": "Chongqing",
  "FJ": "Fujian",
  "GD": "Guangdong",
  "GS": "Gansu",
  "GX": "Guangxi Zhuang Autonomous Region",
  "GZ": "Guizhou",
  "HA": "Henan",
  "HB": "Hubei",
  "HE": "Hebei",
  "HI": "Hainan",
  "HK": "Hong Kong",
  "HL": "Heilongjiang",
  "HN": "Hunan",
  "JL": "Jilin",
  "JS": "Jiangsu",
  "JX": "Jiangxi",
  "LN": "Liaoning",
  "MO": "Macau",
  "NM": "Inner Mongolia",
  "NX": "Ningxia Hui Autonomous Region",
  "QH": "Qinghai",
  "SC": "Sichuan",
  "SD": "Shandong",
  "SH": "Shanghai",
  "SN": "Shaanxi",
  "SX": "Shanxi",
  "TW": "Taiwan Province, People's Republic of China",
  "TW-KEE": "Keelung",
  "XJ": "Xinjiang",
  "XZ": "Tibet Autonomous Region",
  "YN": "Yunnan",
  "ZJ": "Zhejiang",
}
`;

exports[`OpenRegistration Localization should localize city, state, and postcode fields when changing the selected country 3`] = `"City*"`;

exports[`OpenRegistration Localization should localize city, state, and postcode fields when changing the selected country 4`] = `"Postal code*"`;

exports[`OpenRegistration Localization should localize city, state, and postcode fields when changing the selected country 5`] = `"Region*"`;

exports[`OpenRegistration Localization should localize city, state, and postcode fields when changing the selected country 6`] = `
Object {
  "AMA": "Amazonas",
  "ANC": "Áncash",
  "APU": "Apurímac",
  "ARE": "Arequipa",
  "AYA": "Ayacucho",
  "CAJ": "Cajamarca",
  "CAL": "Callao",
  "CUS": "Cusco",
  "HUC": "Huanuco",
  "HUV": "Huancavelica",
  "ICA": "Ica",
  "JUN": "Junín",
  "LAL": "La Libertad",
  "LAM": "Lambayeque",
  "LIM": "Lima",
  "LOR": "Loreto",
  "MDD": "Madre de Dios",
  "MOQ": "Moquegua",
  "PAS": "Pasco",
  "PIU": "Piura",
  "PUN": "Puno",
  "SAM": "San Martín",
  "TAC": "Tacna",
  "TUM": "Tumbes",
  "UCA": "Ucayali",
}
`;

exports[`OpenRegistration Localization should localize city, state, and postcode fields when changing the selected country 7`] = `"District*"`;

exports[`OpenRegistration Localization should localize city, state, and postcode fields when changing the selected country 8`] = `"Postal code*"`;

exports[`OpenRegistration Localization should localize city, state, and postcode fields when changing the selected country 9`] = `
Object {
  "AK": "Alaska",
  "AL": "Alabama",
  "AR": "Arkansas",
  "AS": "American Samoa",
  "AZ": "Arizona",
  "CA": "California",
  "CO": "Colorado",
  "CT": "Connecticut",
  "DC": "District of Columbia",
  "DE": "Delaware",
  "FL": "Florida",
  "GA": "Georgia",
  "GU": "Guam",
  "HI": "Hawaii",
  "IA": "Iowa",
  "ID": "Idaho",
  "IL": "Illinois",
  "IN": "Indiana",
  "KS": "Kansas",
  "KY": "Kentucky",
  "LA": "Louisiana",
  "MA": "Massachusetts",
  "MD": "Maryland",
  "ME": "Maine",
  "MI": "Michigan",
  "MN": "Minnesota",
  "MO": "Missouri",
  "MP": "Northern Mariana Islands",
  "MS": "Mississippi",
  "MT": "Montana",
  "NC": "North Carolina",
  "ND": "North Dakota",
  "NE": "Nebraska",
  "NH": "New Hampshire",
  "NJ": "New Jersey",
  "NM": "New Mexico",
  "NV": "Nevada",
  "NY": "New York",
  "OH": "Ohio",
  "OK": "Oklahoma",
  "OR": "Oregon",
  "PA": "Pennsylvania",
  "PR": "Puerto Rico",
  "RI": "Rhode Island",
  "SC": "South Carolina",
  "SD": "South Dakota",
  "TN": "Tennessee",
  "TX": "Texas",
  "UM": "United States Minor Outlying Islands",
  "UT": "Utah",
  "VA": "Virginia",
  "VI": "United States Virgin Islands",
  "VT": "Vermont",
  "WA": "Washington",
  "WI": "Wisconsin",
  "WV": "West Virginia",
  "WY": "Wyoming",
}
`;

exports[`OpenRegistration Localization should localize city, state, and postcode fields when changing the selected country 10`] = `"State*"`;

exports[`OpenRegistration Localization should localize city, state, and postcode fields when changing the selected country 11`] = `"City*"`;

exports[`OpenRegistration Localization should localize city, state, and postcode fields when changing the selected country 12`] = `"ZIP code*"`;

exports[`OpenRegistration Localization should localize city, state, and postcode fields when changing the selected country 13`] = `
Object {
  "AJ": "Ajman Emirate",
  "AZ": "Abu Dhabi Emirate",
  "DU": "Dubai",
  "FU": "Fujairah",
  "RK": "Ras al-Khaimah",
  "SH": "Sharjah Emirate",
  "UQ": "Umm al-Quwain",
}
`;

exports[`OpenRegistration Localization should localize city, state, and postcode fields when changing the selected country 14`] = `"Emirate*"`;

exports[`OpenRegistration Localization should localize city, state, and postcode fields when changing the selected country 15`] = `
Object {
  " ": " ",
}
`;

exports[`OpenRegistration Localization should localize city, state, and postcode fields when changing the selected country 16`] = `"City*"`;

exports[`OpenRegistration Localization should localize city, state, and postcode fields when changing the selected country 17`] = `"Postal code*"`;

exports[`OpenRegistration Registration should register a new user with a voucher. 1`] = `" <h4 class=\\"llms-notification-title\\">Course enrollment success!</h4> <div class=\\"llms-notification-body\\"><p>Congratulations! You enrolled in LifterLMS Quickstart Course</p> </div> "`;
